yesod-core/src/Yesod/Core/Class/Yesod.hs

```diff
@@ -249,6 +249,16 @@ class RenderRoute site => Yesod site where
     jsAttributes :: site -> [(Text, Text)]
     jsAttributes _ = []
 
+    -- | Same as @jsAttributes@ but allows you to run arbitrary Handler code
+    --
+    -- This is useful if you need to add a randomised nonce value to the script
+    -- tag generated by @widgetFile@. If this function is overridden then
+    -- @jsAttributes@ is ignored.
+    --
+    -- @since 1.6.16
+    jsAttributesHandler :: HandlerFor site [(Text, Text)]
+    jsAttributesHandler = getYesod >>= return . jsAttributes
```

Comment:
This can also be written as `jsAttributes <$> getYesod`, but there difference isn't that improtant.

Reply:
I'm happy to make that change, and also some more hlint-driven cleanup along the way.